code/datums/components/grillable.dm

```diff
@@ -68,6 +68,18 @@
 /datum/component/grillable/proc/FinishGrilling(atom/grill_source)
 
 	var/atom/original_object = parent
+
+	if(istype(parent, /obj/item/stack)) //Check if its a sheet, for grilling multiple things in a stack
+		var/obj/item/stack/itemstack = original_object
+		var/atom/grilled_result = new cook_result(original_object.loc, itemstack.amount)
```

Comment:
i'm going to note that stacks being grilled into non-stacks won't work with this, but someone can lay that groundwork when they get there because there will need to be a lot more considerations.